### DISCUSSION TITLE
access env variable in collection YAML

### DISCUSSION THREAD

**Discussion by CptCompiler** [statamic/cms#6619] [Troubleshooting]
I am trying to set the url for the preview_target for a collection via env variable.

I've tried this: https://v2.statamic.com/environments#option-interpolation (this is for statamic 2)
`{env:FRONTEND_URL}`

and I've tried to read from the config:

config/app.php -> `'frontend_url' => env('FRONTEND_URL', 'http://localhost:3000'),`
and then in .YAML file {config:app:frontend_url}

nothing works
Does anybody know how this is possible?
Thanks

**Comment by edalzell** [comment]
Why does it have to be in the collection yaml?

**Reply by CptCompiler** [reply]
Hi @edalzell !
BTW I’m using statamic headless. 
Because I want to set the preview target with the URL to the frontend. And the URL depends on the environment. 
for local development it’s http://localhost:3000 and for production it’s something else.

**Comment by CptCompiler** [comment]
This is the only solution I have:


Set the config `frontend_url` in `config/app.php`
```
 'frontend_url' => env('FRONTEND_URL', 'http://localhost:3000'),
```

Make a custom tag:

```
<?php

namespace App\Tags;

use Statamic\Tags\Tags;

class CustomConfig extends Tags
{
    /**
     * The {{ custom_config:frontend_url }} tag.
     *
     * @return string|array
     */
    public function frontendUrl()
    {
        return config('app.frontend_url');
    }
}
```

and use it like this in the YAML file:

```
preview_targets:
  - label: Entry
    url: "{{ custom_config:frontend_url }}/api/preview?uri={{uri}}&preview=true"
```

**Comment by edalzell** [comment]
But the front end url has nothing to do with the collection right? It's global?

**Reply by CptCompiler** [reply]
Yes thats correct. I just want to set a different preview target url for different environments.

**Comment by wizzer711** [comment]
Is the custom tag option still supposed to work in Statamic 5?
I am running v5.13.0 and also have the same use case, I tried the custom tag recommendation above but it didn't appear to work.

I would like to pass a env variable into the collection yaml file so I change between local & prod by just using .env variables.

**Reply by edalzell** [reply]
what's your actual use case?

**Reply by wizzer711** [reply]
Very similar to @CptCompiler I think. We are running Statamic as a headless CMS with in our case a Nuxt frontend. We have multiple versions of the CMS, a local version, staging & prod with matching frontend's so it would be good if we could make the preview url dynamic so that the live preview will point at the correct frontend depending on the CMS we are testing from.

As the live_preview url is set in the .yaml file and we are currently tracking that yaml file via git for deployments it would be nice if we could pass an .env variable into the live_preview url so that we can simply change via a .env file.

**Reply by ryanmitchell** [reply]
I've opened a PR here - https://github.com/statamic/cms/pull/10443 - to pass the config to the url generation, this lets you then use this in your targets - eg `{{ config:app.url }}/live-preview`

**Reply by wizzer711** [reply]
thx @ryanmitchell thats awesome, I'll keep an eye on the PR

**Reply by ryanmitchell** [reply]
it just merged so will be in the next release

**Reply by wizzer711** [reply]
That's awesome, thank you for the speed of the pr.
I look forward to the next release.
